src/axom/primal/operators/detail/intersect_impl.hpp

```diff
@@ -1687,6 +1688,305 @@ AXOM_HOST_DEVICE bool intersect_plane_tet3d(const Plane<T, 3>& p,
   return caseNumber > 0 && caseNumber < 15;
 }
 
+/*! \brief Determines if a line intersects a bilinear patch.
+ * \param [in] p0 The first corner of the bilinear patch.
+ * \param [in] p1 The second corner in ccw order.
+ * \param [in] p2 The third corner.
+ * \param [in] p3 The fourth corner.
+ * \param [in] line The line to intersect with the bilinear patch.
+ * \param [out] u The u parameter(s) of the intersection point.
+ * \param [out] v The v parameter(s) of the intersection point.
+ * \param [out] t The t parameter(s) of the intersection point.
+ * \param [in] isRay If true, only return intersections with t >= 0.
+ *
+ * Implements GARP algorithm from Chapter 8 of Ray Tracing Gems (2019)
+ *
+ * \note A bilinear patch is parameterized in [0, 1) x [0, 1) 
+ * 
+ * \warning Always returns false if the line is coplanar to a planar polygon
```

Comment:
Is this a limitation of the algorithm itself?  What if the coplanar line passes through the interior of the polygon?  (Do we care?)

Reply:
It's partially a limitation of the algorithm, although there is a fairly tedious way one could find those intersections via projecting onto that plane and back. At the same time, returning no intersection causes the calling surface-ray intersection routine to return no intersection at tangent points, which is the desired behavior for the winding number routines I need it for.